--- flac2mp3.pl	2008-03-30 15:44:48.000000000 +0100
+++ flac2mp3-quotemeta-patch.pl	2008-03-30 15:41:35.000000000 +0100
@@ -465,24 +465,26 @@
 
     # Building command used to convert file (tagging done afterwards)
     # Needs some work on quoting filenames containing special characters
-    my $quotedsrc  = $srcfilename;
-    my $quoteddest = $destfilename;
+    my $quotedsrc  = quotemeta $srcfilename;
+    my $quoteddest = quotemeta $destfilename;
 
     if ( ( !$pflags{exists} || $pflags{md5} || $::Options{force} )
         && !$::Options{tagsonly} )
     {
 
         $::Options{info}
-          && msg( $pretendString . "Transcoding \"$quotedsrc\"" );
+          && msg( $pretendString . "Transcoding \"$srcfilename\"" );
 
         # Transcode to a temp file in the destdir.
         # Rename the file if the conversion completes sucessfully
         # This avoids leaving incomplete files in the destdir
         # If we're "pretending", don't create a File::Temp object
         my $tmpfilename;
+        my $quotedtmp;
         my $tmpfh;
         if ( $::Options{pretend} ) {
-            $tmpfilename = $quoteddest;
+            $tmpfilename = $destfilename;
+            $quotedtmp = $quoteddest;
         }
         else {
 
@@ -497,11 +499,12 @@
                 SUFFIX => '.tmp'
             );
             $tmpfilename = $tmpfh->filename;
+            $quotedtmp = quotemeta $tmpfilename;
         }
 
         my $convert_command =
-            "\"$flaccmd\" @flacargs \"$quotedsrc\""
-          . "| \"$lamecmd\" @lameargs - \"$tmpfilename\"";
+            "\"$flaccmd\" @flacargs $quotedsrc"
+          . "| \"$lamecmd\" @lameargs - $quotedtmp";
 
         $::Options{debug} && msg("$convert_command");
 
